flytekit/core/array_node.py

```diff
@@ -27,7 +31,7 @@
 class ArrayNode:
     def __init__(
         self,
-        target: LaunchPlan,
+        target: Any,
```

Comment:
Can we use `Union` to describe the types? I'd rather keep `ArrayNode` well-typed.

Reply:
yup - getting an update for this. Was working around a circular imports error. Can use if TYPE_CHECKING